data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuditMessage.java

```diff
@@ -97,18 +114,31 @@ protected AuditMessage(
     }
 
     @Generated
+    public String getLeafResourceId() {
+        return leafResourceId;
+    }
+
+    @Generated
+    public String getToken() {
+        return token;
+    }
+
+    @Generated
+    @Nullable
     public String getUserId() {
         return userId;
     }
 
     @Generated
+    @Nullable
     public String getResourceId() {
         return resourceId;
     }
 
     @Generated
-    public Context getContext() throws JsonProcessingException {
-        return MAPPER.treeToValue(context, Context.class);
+    @Nullable
+    public Context getContext() {
+        return context;
```

Comment:
Return `Optional`.

Reply:
Will raise as ticked to cover this across all of the services